pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -394,22 +395,22 @@ func (c *Reconciler) updateTaskRunsStatusDirectly(pr *v1alpha1.PipelineRun) erro
 }
 
 func (c *Reconciler) createTaskRun(logger *zap.SugaredLogger, rprt *resources.ResolvedPipelineRunTask, pr *v1alpha1.PipelineRun, storageBasePath string) (*v1alpha1.TaskRun, error) {
-	var taskRunTimeout = &metav1.Duration{Duration: 0 * time.Second}
+	var taskRunTimeout = &metav1.Duration{Duration: apisconfig.NoTimeoutDuration}
 
-	if pr.Spec.Timeout != nil {
+	// If the value of the timeout is 0 for any resource, there is no timeout.
+	// It is impossible for pr.Spec.Timeout to be nil, since SetDefault always assigns it with a value.
+	if pr.Spec.Timeout.Duration != apisconfig.NoTimeoutDuration {
 		pTimeoutTime := pr.Status.StartTime.Add(pr.Spec.Timeout.Duration)
 		if time.Now().After(pTimeoutTime) {
 			// Just in case something goes awry and we're creating the TaskRun after it should have already timed out,
-			// set a timeout of 0.
+			// set the timeout to 1 second.
 			taskRunTimeout = &metav1.Duration{Duration: time.Until(pTimeoutTime)}
 			if taskRunTimeout.Duration < 0 {
-				taskRunTimeout = &metav1.Duration{Duration: 0 * time.Second}
+				taskRunTimeout = &metav1.Duration{Duration: 1 * time.Second}
```

Comment:
the more I look at this, I'm not really sure why this case is being explicitly handled at all - the way that the creation works, we could easily hit a case where by the time a `TaskRun` is actually created, the current time is after `pTimeoutTime`, even if this case isn't hit 🤔 

But I'm fine with leaving this as-is in this PR.